src/controllers/commentController.ts

```diff
@@ -60,4 +61,56 @@ export const getComments = async (
   } catch (error) {
     return next(error);
   }
+};
+
+export const deleteCommentToBook = async (
+  req: Request<{ commentId: string }, {}, {}, { bookId?: string }>,
+  res: Response,
+  next: NextFunction
+) => {
+  const { commentId } = req.params;
+
+  const user: IUser | undefined = req.user;
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+  const comment = await Comment.findById(commentId);
```

Comment:
another approach could be something like(pseudocode):
find({
id: commentId,
user: user.userId
})

if not found("Comment not found for this user).
the `user.userId == comment.user' i.e lines 84-88 won't be necessary anymore.